index.js

```diff
@@ -41,7 +41,7 @@ var Client = function (host, opts) {
 
   if (!opts) opts = {}
 
-  var disc = opts.token || /^https:\/\/discovery.etcd.io\//.test(host || '')
+  var disc = opts.token || /^https:\/\/discovery\.etcd\.io\//.test(host || '')
```

Comment:
```suggestion
  var disc = opts.token || host?.startsWith('https://discover.etcd.io/')
```

Reply:
turns out this introduces a breaking change when host is an array (this came up in tests with furthermore). Can we use .test or is there an alternative that allows us to check if host is an array ?

Reply:
Yikes.  It was working on accident before.  `.test` explicitly takes a string, and was casting the array to a string under the hood.

It'd be best if we had code above this to make sure `opts` is the exact type we want.  Typically that's done by checking if it's a string, and then moving it into an array.

Then you can use `.some` with `.startsWith` e.g.

```javascript
if (typeof host === 'string') {
  host = [host]
}
host.some(h => h.startsWith('foo'))
```

Reply:
perf! thanks for the suggestion with greater context on what it **should** be doing